fastcrypto-tbls/src/polynomial.rs

```diff
@@ -29,18 +30,29 @@ pub type PublicPoly<C> = Poly<C>;
 
 /// Vector related operations.
 
-impl<C> Poly<C> {
+impl<C: GroupElement> Poly<C> {
     /// Returns the degree of the polynomial
     pub fn degree(&self) -> usize {
         // e.g. c_0 + c_1 * x + c_2 * x^2 + c_3 * x^3
         // ^ 4 coefficients correspond to a 3rd degree poly
         self.0.len() - 1
     }
+
+    fn reduce(&mut self) {
+        while self.0.len() > 1 && self.0.last() == Some(&C::zero()) {
+            self.0.pop();
+        }
+    }
 }
 
-impl<C> From<Vec<C>> for Poly<C> {
+impl<C: GroupElement> From<Vec<C>> for Poly<C> {
     fn from(c: Vec<C>) -> Self {
-        Self(c)
+        if c.is_empty() {
+            return Self::zero();
+        }
+        let mut p = Self(c);
+        p.reduce();
+        p
```

Comment:
@benr-ml This changes the existing implementation, but I think it's okay. The `degree` function is at least not working correctly until this is done.

Reply:
this change should be behind a flag when merged to sui, since otherwise an attacker that sends a message with a polynomial that is reduced can cause a fork between validators that run the new code vs ones that run the old code